core/src/main/java/apoc/date/Date.java

```diff
@@ -162,7 +163,7 @@ public Long fromISO8601(final @Name("time") String time) {
 	@UserFunction
 	@Description("apoc.date.parse('2012-12-23','ms|s|m|h|d','yyyy-MM-dd') - parse date string using the specified format into the specified time unit")
 	public Long parse(@Name("time") String time, @Name(value = "unit", defaultValue = "ms") String unit, @Name(value = "format",defaultValue = DEFAULT_FORMAT) String format, final @Name(value = "timezone", defaultValue = "") String timezone) {
-		Long value = parseOrThrow(time, getFormat(format, timezone));
+		Long value = StringUtils.isBlank(time) ? null : parseOrThrow(time, getFormat(format, timezone));
```

Comment:
Should we use isEmpty() instead of isBlank()? Wondering if we should fail when the string is whitespace only. What do you think?

Reply:
@vga91 did with `isEmpty`, but I asked to change it to `isBlank` because given my experience as data engineer the data can be very very very messy, so in order to be more lenient I asked to him to do in this way.
@JMHReif please lemme know if you prefer having `isEmpty`